editoast/src/schema/v2/trainschedule.rs

```diff
@@ -0,0 +1,516 @@
+pub enum PathItemLocation {
+    TrackOffset {
+        /// The track section ID
+        #[schema(inline)]
+        track: NonBlankString,
+        /// The offset in millimeters from the start of the track
+        offset: u64,
+    },
+    OperationalPointId {
+        /// The object id of an operational point
+        #[schema(inline)]
+        operational_point: NonBlankString,
+    },
+    OperationalPointDescription {
+        /// The operational point trigram
+        #[schema(inline)]
+        trigram: NonBlankString,
+        /// An optional secondary code to identify a more specific location
+        secondary_code: Option<String>,
+    },
+    OperationalPointUic {
+        /// The [UIC](https://en.wikipedia.org/wiki/Railway_vehicle_owner%27s_code) code of an operational point
+        uic: u32,
+        /// An optional secondary code to identify a more specific location
+        secondary_code: Option<String>,
+    },
+}
+
+/// A location on the path of a train
+#[derive(Debug, Clone, Serialize, Deserialize, ToSchema)]
+pub struct PathItem {
+    /// The unique identifier of the path item.
+    /// This is used to reference path items in the train schedule.
+    #[schema(inline)]
+    pub id: NonBlankString,
+    /// Metadata given to mark a point as wishing to be deleted by the user.
+    /// It's useful for soft deleting the point (waiting to fix / remove all references)
+    /// If true, the train schedule is consider as invalid and must be edited
+    #[serde(default)]
+    pub deleted: bool,
+    #[serde(flatten)]
+    #[schema(inline)]
+    pub location: PathItemLocation,
+}
+
+#[derive(Debug, Clone, Serialize, Derivative, ToSchema)]
+#[serde(deny_unknown_fields)]
+#[derivative(Default)]
+pub struct Margins {
+    #[schema(inline)]
+    pub boundaries: Vec<NonBlankString>,
+    #[derivative(Default(value = "vec![MarginValue::Zero]"))]
+    /// The values of the margins. Must contains one more element than the boundaries
+    /// Can be a percentage `X%`, a time in minutes per kilometer `Xmin/km` or `0`
+    #[schema(value_type = Vec<String>, example = json!(["0", "5%", "2min/km"]))]
+    pub values: Vec<MarginValue>,
+}
+
+impl<'de> Deserialize<'de> for Margins {
+    fn deserialize<D>(deserializer: D) -> Result<Margins, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        struct InternalMargins {
+            boundaries: Vec<NonBlankString>,
+            values: Vec<MarginValue>,
+        }
+
+        let InternalMargins { boundaries, values } = InternalMargins::deserialize(deserializer)?;
+        if boundaries.len() + 1 != values.len() {
+            return Err(serde::de::Error::custom(
+                "The number of boudaries and values must be the same",
+            ));
+        }
+        Ok(Margins { boundaries, values })
+    }
+}
+
+#[derive(Debug, Copy, Clone, Default, PartialEq)]
+pub enum MarginValue {
+    #[default]
+    Zero,
```

Comment:
Why do we need this special case? Can't `Percentage(0.)` work?

Reply:
`0` means no margins were defined. The frontend should handle this special case, it's a way to explicit this case.

Reply:
We will rename the variant `None` and it serialize as `none`